test/src/expectEvent.truffle.test.js

```diff
@@ -463,12 +525,163 @@ contract('expectEvent (truffle contracts)', function ([deployer]) {
+          });
+        });
+      });
+
+      context('with indirectly called contract', function () {
+        it('accepts events emitted from other contracts with emitter object', async function () {
+          expectEvent.inTransaction(this.txHash, this.secondEmitter, 'IndexedUint', {
+            indexedValue: this.indexedValue2,
+            normalValue: this.normalValue2,
+          });
+        });
+
+        it('accepts emitted indexed events with emitter class', async function () {
+          expectEvent.inTransaction(this.txHash, IndirectEventEmitter, 'IndexedUint', {
+            indexedValue: this.indexedValue2,
+            normalValue: this.normalValue2,
+          });
+        });
+      });
+    });
+
+    describe('with conflicting indexed event parameters', function () {
+      beforeEach(async function () {
+        this.indexedValue1 = new BN(42);
+        this.normalValue1 = new BN(2014);
+        this.indexedValue2 = new BN(2016);
+        this.indexedValue3 = new BN(2009);
+        const { receipt } = await this.emitter.emitIndexedConflictingUint(
+          this.indexedValue1,
+          this.normalValue1,
+          this.indexedValue2,
+          this.indexedValue3,
+          this.secondEmitter.address
+        );
+        this.txHash = receipt.transactionHash;
+      });
+
+      context('with directly called contract', function () {
+        it('accepts emitted events with correct indexed parameter', async function () {
+          await expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedConflictingUint', {
+            indexedValue1: this.indexedValue1,
+            normalValue1: this.normalValue1,
+          });
+        });
+
+        it('throws if the emitter class is passed', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, EventEmitter, 'IndexedConflictingUint', {
+            indexedValue1: this.indexedValue1,
+            normalValue1: this.normalValue1,
+          }));
+        });
+
+        it('throws if the event value emitted from other contract is passed', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedConflictingUint', {
+            indexedValue1: this.indexedValue2,
+            normalValue1: this.indexedValue3,
+          }));
+        });
+
+        it('throws if the event emitted from other contract is passed', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, this.emitter, 'IndexedConflictingUint', {
+            normalValue: this.normalValue,
+            indexedValue2: this.indexedValue2,
+          }));
+        });
+
+        it('throws if the wrong event is requested', async function () {
+          await assertFailure(expectEvent.inTransaction(this.txHash, this.secondEmitter, 'IndexedConflictingUint', {
+            indexedValue2: this.indexedValue1,
+            indexedValue3: this.normalValue1,
```

Comment:
See here, this is a test to ensure that expectEvent is not accepting events if using the ABI of another. I was not sure if this is really necessary, but I added it as it looks a plausible test to be done for checking if the events are being returned correctly.